### PR TITLE
Fixes #9026: cfengine stops processing a promise on symlinks

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9026

### CODE REVIEW

**Comment by amousset** [comment]
@peckpeck Need rebasing.

**Comment by peckpeck** [comment]
PR rebased

**Comment by amousset** [comment]
@peckpeck Need rebasing again!

**Comment by peckpeck** [comment]
PR rebased

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: rudder-agent/SOURCES/patches/cfengine/31-fix-symlink-traversal.patch [hunk 1]
```diff
@@ -33,4 +34,26 @@ diff -rNu cfengine-source/libutils/file_lib.c cfengine-source-new/libutils/file_
...
++    uid_t p_uid = geteuid();
+ 
+     if ((dirfd = safe_open(parent_dir, O_RDONLY)) == -1)
+     {
+@@ -702,7 +709,10 @@ static int safe_open_true_parent_dir(const char *path,
+         goto cleanup;
+     }
+ 
+-    if (traversed_link && (link_user != statbuf.st_uid || link_group != statbuf.st_gid))
++    if ( traversed_link &&
```
Comment (amousset) [inline rudder-agent/SOURCES/patches/cfengine/31-fix-symlink-traversal.patch]: Could you add the comment here too?

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**